client/scripts/views/pages/discussions/index.ts

```diff
@@ -67,41 +66,85 @@ const DiscussionsPage: m.Component<{ topic?: string }, IDiscussionPageState> = {
-      ? Number(localStorage[`${app.activeId()}-lookback`])
-      : Number.isInteger(vnode.state.lookback)
-        ? vnode.state.lookback
-        : vnode.state.defaultLookback;
+    vnode.state.lookback[subpage] = (returningFromThread && localStorage[`${app.activeId()}-lookback-${subpage}`])
+      ? localStorage[`${app.activeId()}-lookback-${subpage}`]
+      : vnode.state.lookback[subpage]?._isAMomentObject
+        ? vnode.state.lookback[subpage]
+        : moment();
+    vnode.state.topicInitialized = {};
   },
   view: (vnode) => {
     const { topic } = vnode.attrs;
     const activeEntity = app.community ? app.community : app.chain;
+    const subpage = topic || 'allProposals';
+
     // add chain compatibility (node info?)
     if (!activeEntity?.serverLoaded) return m(PageLoading, {
       title: topic || 'Discussions',
       showNewProposalButton: true,
     });
 
-    localStorage[`${app.activeId()}-lookback`] = vnode.state.lookback;
+    if (!vnode.state.topicInitialized[subpage]) {
+      vnode.state.postsDepleted = false;
+      $(window).off('scroll');
+
+      // Fetch first page of posts
+      if (!vnode.state.lookback[subpage]
+        || !vnode.state.lookback[subpage]?._isAMomentObject) {
+        vnode.state.lookback[subpage] = moment();
+      }
+      let topicId;
+      if (topic) {
+        topicId = app.topics.getByName(topic, app.activeId())?.id;
+        if (!topicId) {
+          notifyError('Cannot fetch posts: Improperly configured topics.');
+          return m(EmptyTopicPlaceholder, {
+            communityName: app.activeId(),
+            topicName: topic
+          });
+        }
+      }
+      const options = {
+        chainId: app.activeChainId(),
+        communityId: app.activeCommunityId(),
+        cutoffDate: vnode.state.lookback[subpage],
+        topicId,
+      };
+      app.threads.loadNextPage(options).then((morePostsRemaining) => {
+        if (!morePostsRemaining) vnode.state.postsDepleted = true;
+        m.redraw();
+      });
+
+      // Initialize infiniteScroll
+      vnode.state.onscroll = _.debounce(async () => {
+        if (vnode.state.postsDepleted) return;
+        const options_ = {
+          chainId: app.activeChainId(),
+          communityId: app.activeCommunityId(),
+          cutoffDate: vnode.state.lookback[subpage],
+          topicId,
+        };
+        const scrollHeight = $(document).height();
+        const scrollPos = $(window).height() + $(window).scrollTop();
+        if (scrollPos > (scrollHeight - 400)) {
+          const morePostsRemaining = await app.threads.loadNextPage(options_);
+          if (!morePostsRemaining) vnode.state.postsDepleted = true;
+          m.redraw();
+        }
+      }, 400);
+
+      $(window).on('scroll', vnode.state.onscroll);
+
+      vnode.state.topicInitialized[subpage] = true;
+    }
+
+    localStorage[`${app.activeId()}-lookback-${subpage}`] = vnode.state.lookback[subpage];
```

Comment:
why is this outside the topicInitialized check?

Reply:
Because the state's lookback gets constantly updated as user scrolls, and we need to ensure that it is up to date if a user changes topic views. If it was in the initialization, only the initial lookback would be saved.